### ISSUE TITLE
define minimal service offer schema

### ISSUE THREAD

**Issue by loomis** [slipstream/SlipStreamServer#940]
Although the schema for ServiceOffer resources is intentionally open, we need to have a minimum schema defined to allow SlipStream to provision resources correctly.  

Define the minimum schema that should be implemented by the "scrapers" and then used by the cloud connectors for provisioning.

**Comment by loomis** [comment]
See the [updated [schema](https://github.com/SixSq/SlipStreamPricing/blob/master/Schema.md).  It still must be verified that it corresponds to the schema that is implicitly in the code.

**Comment by loomis** [comment]
The schema has been updated after working with the AWS pricing information.  Further comments welcome.

**Comment by loomis** [comment]
An example ServiceOffer document for AWS is:
```
{
 "name":        "17/117000/48000 [US] (hs1.8xlarge)",
 "description": "VM (standard) with 17 vCPU, 117000 MiB RAM, 48000 GB disk [US] (hs1.8xlarge)",

 "resource:vcpu":     17,
 "resource:ram":      117000,
 "resource:disk":     48000,
 "resource:diskType": "HDD",
 "resource:type":     "VM",
 "resource:class":    "standard",
 "resource:platform": "aws",
 "resource:country":  "US"

 "price:unitCost":          4.367226810975031,
 "price:billingPeriodCode": "MON",
 "price:currency":          "EUR",
 "price:billingUnitCode":   "HUR",
 "price:unitCode":          "HUR",
 "price:freeUnits":         0,

 "aws:operatingSystem": "Linux",
 "aws:sku":             "64EUYDYDPNWKXQUX",
 "aws:instanceType":    "hs1.8xlarge",
 }
```

**Comment by mebster** [comment]
How does this offer refer to a given cloud connector?  Is there a container document that includes hers to these?  Or is there another mechanism for these associations?

**Comment by loomis** [comment]
@mebster Yes, there is a field `connector/href` that provides the reference to the connector. This is defined in the [schema](https://github.com/SixSq/SlipStreamPricing/blob/master/Schema.md) but not shown in the example. I just realized that the region isn't in the example; it should be part of the 'aws' attributes.

**Comment by loomis** [comment]
Here's an updated example, with the region information added and the reference to the connector.  The value for the reference will eventually change from "ec2-eu-central-1" to "connector/ec2-eu-central-1" once connectors become independent resources. It's an open question for me whether to define another field with just the simple name of the connector. 

```
{
  "connector": {"href": "ec2-eu-central-1"},
  "name": "4/15000/80 [DE] (m3.xlarge)",
  "description": "VM (standard) with 4 vCPU, 15000 MiB RAM, 80 GB disk [eu-central-1, DE] (m3.xlarge)",

  "resource:vcpu": 4,
  "resource:ram": 15000,
  "resource:disk": 80,
  "resource:diskType": "SSD",
  "resource:type": "VM",
  "resource:class": "standard",
  "resource:country": "DE",
  "resource:platform": "aws",

  "price:unitCost": 0.0,
  "price:unitCode": "HUR",
  "price:freeUnits": 0,
  "price:currency": "EUR",
  "price:billingPeriodCode": "MON",
  "price:billingUnitCode": "HUR",

  "aws:location": "EU (Frankfurt)",
  "aws:region": "eu-central-1",
  "aws:instanceType": "m3.xlarge",
  "aws:sku": "2EBFV7TK7GHEPNPW",
  "aws:operatingSystem": "Linux"
}
```

**Comment by loomis** [comment]
I've added draft resources for ServiceAttributeNamespace and ServiceAttribute to Nuvla to have a better idea how the various resources that are part of the service catalog will interact.

**Comment by loomis** [comment]
This is being implemented in sixsq/tasklist#889.  Updates to the schema will be tracked in the [schema documentation](https://github.com/SixSq/SlipStreamPricing/blob/master/Schema.md).
